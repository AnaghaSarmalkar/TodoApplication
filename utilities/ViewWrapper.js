import React from "react";
import { View } from "react-native";

export default function ViewWrapper(props) {
  return (
    <View
      {...props}
      // style={{
      //   flex: 1,
      //   alignItems: "center",
      //   justifyContent: "center",
      //   // paddingTop: insets.top,
      // }}
    >
      {props.children}
    </View>
  );
}
