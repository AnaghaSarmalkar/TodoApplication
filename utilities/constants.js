import { Platform } from "react-native";

export const isIos = Platform.OS === "ios";
export const isWeb = Platform.OS === "web";
export const isAndroid = Platform.OS === "android";
