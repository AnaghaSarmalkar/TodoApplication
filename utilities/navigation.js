export function setPageTitle(navigation, title) {
  if (title) {
    navigation.setOptions({
      title: title,
    });
  }
}
