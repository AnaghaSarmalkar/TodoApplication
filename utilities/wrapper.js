import React from "react";
import { SafeAreaView } from "react-native-safe-area-context";

function SafeAreaWrapper({ children }) {
  <SafeAreaView>{children}</SafeAreaView>;
}
