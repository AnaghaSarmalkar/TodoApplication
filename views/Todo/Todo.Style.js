import { StyleSheet } from "react-native";

export const styles = StyleSheet.create({
  container: {
    display: "flex",
    alignItems: "center",
    flexDirection: "column",
    flex: 1,
    paddingLeft: 20,
    paddingRight: 20,
    paddingTop: 20,
  },
  content: {
    flex: 1,
  },
  header: {
    paddingBottom: 20,
  },
  footer: {
    display: "flex",
    flexDirection: "row",
    width: "100%",
    justifyContent: "space-around",
    marginBottom: 0,
  },
});
