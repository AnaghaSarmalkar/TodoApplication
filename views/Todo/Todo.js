import React from "react";
import { SafeAreaView } from "react-native-safe-area-context";
import Error from "../../components/Error/Error";
import Header from "../../components/Header/Header";
import Loading from "../../components/Loading/Loading";
import TextArea from "../../components/TextArea/TextArea";

export default function Todo({ loading, error, data }) {
  if (loading) return <Loading />;
  if (error) return <Error message={error.message} />;
  return (
    <SafeAreaView>
      <Header title={data.locations.name}></Header>
      <TextArea description={data.locations.description}></TextArea>
    </SafeAreaView>
  );
}
