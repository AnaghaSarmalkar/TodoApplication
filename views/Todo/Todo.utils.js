export function onNavButtonPress(navigation, navigationScreen) {
  navigation.navigate(navigationScreen);
}
