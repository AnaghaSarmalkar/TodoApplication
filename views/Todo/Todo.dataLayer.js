import { gql, useQuery } from "@apollo/client";
import React from "react";
import Todo from "./Todo";

export const getLocationsDesc = gql`
  query GetLocationsDesc {
    locations {
      id
      name
      description
    }
  }
`;

export default function TodoDataLayer() {
  const { loading, error, data } = useQuery(getLocationsDesc);

  return <Todo loading={loading} error={error} data={data} />;
}
