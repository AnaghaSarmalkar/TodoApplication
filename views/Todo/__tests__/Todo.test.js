import React from "react";
import { shallow } from "../../../utilities/testing";
import Todo from "../Todo";

describe("Todo", () => {
  it("renders loading component", async () => {
    const wrapper = shallow(
      <Todo data={undefined} loading={true} error={undefined} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("renders error component", async () => {
    const error = new Error("An error occured.");
    const wrapper = shallow(
      <Todo data={undefined} loading={false} error={error} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("renders List component", async () => {
    const data = {
      data: {
        locations: [
          {
            id: "1",
            name: "Test Location 1",
            description: "Test Description 1",
          },
          {
            id: "2",
            name: "Test Location 2",
            description: "Test Description 2",
          },
          {
            id: "3",
            name: "Test Location 3",
            description: "Test Description 3",
          },
        ],
      },
    };

    const wrapper = shallow(
      <Todo data={data} loading={false} error={undefined} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
