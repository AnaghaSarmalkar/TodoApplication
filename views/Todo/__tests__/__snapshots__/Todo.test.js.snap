// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo renders error component 1`] = `
<Error
  message="An error occured."
/>
`;

exports[`Todo renders loading component 1`] = `<Loading />`;
